### DISCUSSION TITLE
External Auth Feature in Kong IC

### DISCUSSION THREAD

**Discussion by mawe91** [Kong/kubernetes-ingress-controller#4605] [Q&A] [answered]
Hi all,
I want to integrate an external authorizer to leave the authentication of a path to an external service. In other ingress controllers there are plugins like this one referencing traefik:
https://doc.traefik.io/traefik/middlewares/http/forwardauth
The first lines on the traffic docs basically explains the use case best:

> The ForwardAuth middleware delegates authentication to an external service. If the service answers with a 2XX code, access is granted, and the original request is performed. Otherwise, the response from the authentication server is returned.

 Is it possible to achieve something similar in kong too without creating my own plugin?

**Comment by programmer04** [comment, marked as answer]
Currently, Kong does not offer an official solution for such a use case, but you can check:
- https://github.com/aunkenlabs/kong-external-auth (community plugin that seems to do what required)
- it sounds similar to what [Open Policy Agent](https://www.openpolicyagent.org/) provides, you can check:
    - https://docs.konghq.com/hub/kong-inc/opa (the official Kong Enterprise plugin)
    - https://github.com/TravelNest/kong-authorization-opa (community plugin)

Please remember that community plugins aren't officially endorsed by Kong. You can also write your own plugin, read more in [the official docs](https://docs.konghq.com/gateway/latest/plugin-development).

> _If the above answers your question please click `Mark as answer` button to increase visibility for others._
